mirdata/datasets/four_way_tabla.py

```diff
@@ -0,0 +1,241 @@
+
+        """
+        return load_onsets(self.onsets_b_path)
+
+    @property
+    def onsets_d(self) -> Optional[annotations.BeatData]:
+        """Onsets for stroke D
+
+        Returns:
+            * annotations.BeatData - onsets annotation
+
+        """
+        return load_onsets(self.onsets_d_path)
+
+    @property
+    def onsets_rb(self) -> Optional[annotations.BeatData]:
+        """Onsets for stroke RB
+
+        Returns:
+            * annotations.BeatData - onsets annotation
+
+        """
+        return load_onsets(self.onsets_rb_path)
+
+    @property
+    def onsets_rt(self) -> Optional[annotations.BeatData]:
+        """Onsets for stroke RT
+
+        Returns:
+            * annotations.BeatData - onsets annotation
+
+        """
+        return load_onsets(self.onsets_rt_path)
+
+    def to_jams(self):
+        """Get the track's data in jams format
+
+        Returns:
+            jams.JAMS: the track's data in jams format
+
+        """
+        return jams_utils.jams_converter(
+            audio_path=self.audio_path,
+            beat_data=[
+                (self.onsets_b, "onsets_b"),
+                (self.onsets_d, "onsets_d"),
+                (self.onsets_rb, "onsets_rb"),
+                (self.onsets_rt, "onsets_rt"),
+            ],
+            metadata={"train": self.train},
+        )
+
+
+@io.coerce_to_bytes_io
+def load_audio(fhandle: BinaryIO) -> Tuple[np.ndarray, float]:
+    """Load a Mridangam Stroke Dataset audio file.
+
+    Args:
+        fhandle (str or file-like): File-like object or path to audio file
+
+    Returns:
+        * np.ndarray - the mono audio signal
+        * float - The sample rate of the audio file
+    """
+    return librosa.load(fhandle, sr=44100, mono=True)
+
+
+@io.coerce_to_string_io
+def load_onsets(fhandle):
+    """Load stroke onsets
+
+    Args:
+        fhandle (str or file-like): Local path where the pitch annotation is stored.
+
+    Returns:
+        EventData: onset annotations
+
+    """
+    onsets = []
+    reader = csv.reader(fhandle, delimiter="\n")
+    for line in reader:
+        onsets.append(float(line[0]))
+
+    if not onsets:
+        return None
+    onsets = np.array(onsets)
+
+    # All strokes are out of an annotated metric
+    beat_position = np.zeros(onsets.shape)
+
+    return annotations.BeatData(onsets, "s", beat_position, "global_index")
+
+
+@core.docstring_inherit(core.Dataset)
+class Dataset(core.Dataset):
+    """
+    The Four-Way Tabla dataset
+    """
+
+    def __init__(self, data_home=None, version="default"):
+        super().__init__(
+            data_home,
+            version,
+            name="four_way_tabla",
+            track_class=Track,
+            bibtex=BIBTEX,
+            indexes=INDEXES,
+            remotes=REMOTES,
+            license_info=LICENSE_INFO,
+        )
+
+    def load_audio(self, *args, **kwargs):
+        return load_audio(*args, **kwargs)
+
+    def load_onsets(self, *args, **kwargs):
```

Comment:
weirdly this also looks not covered by tests although I see that you do test load_onsets in test_get_onsets